cli/upload.ts

```diff
@@ -96,17 +97,23 @@ export async function handleUpload({
     printAndExit(`Could not find a Pack id registered in directory "${manifestDir}"`);
   }
 
-  const packVersion = manifest.version;
+  const metadata = compilePackMetadata(manifest);
+  let packVersion = manifest.version;
   if (!packVersion) {
-    printAndExit(`No Pack version declared for this Pack`);
+    const nextPackVersionInfo = await client.getNextPackVersion(
+      packId,
+      {},
+      {proposedMetadata: JSON.stringify(metadata)},
+    );
+    packVersion = nextPackVersionInfo.nextVersion;
```

Comment:
what's the potential issues with not having the version in the metadata? I guess I'm surprised because this metadata will be assoc. with the version in the db

Reply:
Good question. You are probably right that we can make it work without a version in metadata but then there's no way to identify the version from a pack upload archive.

I did some code search that in order to remove version from metadata you'll just need to clean up the methods with the assumption that metadata has version. So far I can only find a few pack-diff functions, but we'd need more testing to see if there's any other dependency.

Reply:
makes sense, thanks for digging in. I'll keep this here and possibly investigate removing it as a followup task

Reply:
Yeah it's mostly that for historical reasons the metadata object from S3 gets passed around in its entirety and some code reads the version from there. I'd say it's not a priority to change that. Let's just make sure it gets set correctly and not worry about it otherwise for now.